docs/src/getting-started/train_yaml_config.rst

```diff
@@ -0,0 +1,378 @@
+.. _train_yaml_config:
+
+Training YAML Reference
+************************************
+Overview
+===================
+``metatrain`` uses a YAML file to specify the parameters for model training,
+accessed via ``mtt train options.yaml``. In this section, we provide a complete reference
+for the parameters provided by the training YAML input.
+
+The YAML input file can be divided into five sections:
+
+- Computational Parameters
+- wandb integration
+- ``architecture``
+- ``loss``
+- Data
+
+Computational Parameters
+======================================
+The computational parameters define the computational device, precision and seed. These parameters are optional.
+
+.. code-block:: yaml
+
+    device: 'cuda'
+    precision: 32
+    seed: 0
+
+:param device [optional]: The computational device used for model training. The script automatically
+    chooses the best option by default. The possible devices that can be used, and the best device option,
+    depend on the model architecture. The easiest way to use this parameter is to use either either ``cpu``, ``gpu``,
+    ``multi-gpu``. Internally, under the choice ``gpu``, the script will automatically choose between ``cuda`` or ``mps``.
+:param precision [optional]: The base precision for all floats in model training. This impacts the datatype used. Possible
+    options are the integers ``64``, ``32`` and ``16``, resulting in the datatype used to be ``float64``, ``float32`` and
+    ``float16`` respectively. The datatypes that can be supported also depends on the model architecture used.
+:param seed [optional]: The seed used for non-deterministic operations and is used to set the seed for ``numpy.random``,
+    ``random``, ``torch`` and ``torch.cuda``. The input must be a non-negative integer. This parameter is important for ensuring
+    reproducibility. If not specified, the seed is generated randomly.
+
+wandb integration
+===================
+The next set of parameters are also optional and deals with integration with Weights and Biases (wandb) logging. Leaving this
+section blank will simply disable wandb integration. The parameters for this section is the same as that in
+`wandb.init <https://docs.wandb.ai/ref/python/init/>`_. Here we provide a minimal example for the YAML input
```

Comment:
I thunk this can be shortened a bit maybe along the lines of:

```suggestion
Optional section dealing with integration with `Weights and Biases (wandb) <link>`_ logging. Leaving this
section blank will simply disable wandb integration. The parameters for this section is the same as that in
`wandb.init <https://docs.wandb.ai/ref/python/init/>`_. Here we provide a minimal example for the YAML input
```